infra-gen2/tool/deploy_gen2.dart

```diff
@@ -0,0 +1,391 @@
+
+  /// cache the config file contents to create environments map
+  final rawConfigContent = amplifyOutputs.readAsStringSync();
+  RegExp exp = RegExp(r"'''(.*?)'''", dotAll: true);
+  final configMap = exp.firstMatch(rawConfigContent)?.group(1) ?? '';
+  amplifyEnvironments[backendName] = '\'\'\'$configMap\'\'\'';
+}
+
+/// Append the environments to amplify_outputs.dart
+void _appendEnvironments(
+  Map<String, String> amplifyEnvironments,
+  AmplifyIntegCategory category,
+  File amplifyOutputs,
+) {
+  /// Skip if there are no amplifyEnvironments
+  if (amplifyEnvironments.isEmpty) {
+    return;
+  }
+  print('👷 Building amplifyEnvironments');
+  var amplifyEnvironmentsOutput =
+      '\nconst amplifyEnvironments = <String, String>{\n';
+  amplifyEnvironments.forEach((key, value) {
+    amplifyEnvironmentsOutput += "'$key': $value,\n";
+  });
+  amplifyEnvironmentsOutput += '};';
+
+  /// Append amplifyEnvironments to amplify_outputs.dart
+  amplifyOutputs.writeAsStringSync(amplifyEnvironmentsOutput,
+      mode: FileMode.append);
+}
+
+/// Copy the amplify_outputs.dart file to other shared paths
+void _copyAmplifyOutputs(
+  List<String> outputPaths,
+  File amplifyOutputs,
+) {
+  if (outputPaths.length <= 1) {
+    return;
+  }
+
+  print('👯 Copying amplify_outputs.dart to other shared paths');
+  for (final outputPath in outputPaths) {
+    final destination = p.join(repoRoot.path, outputPath);
+    final outputFile = File(p.join(destination, 'amplify_outputs.dart'));
+
+    if (!outputFile.existsSync()) {
+      outputFile.createSync(recursive: true);
+    }
+    final amplifyOutputsContents = amplifyOutputs.readAsStringSync();
+
+    outputFile.writeAsStringSync(amplifyOutputsContents);
+  }
+}
+
+/// Create a unique bucket name
+String _createBucketName(String base) {
+  String uniqueShort = uuid().substring(0, 8);
+  return base.toLowerCase() + '-gen2-integ-' + uniqueShort;
+}
+
+String _getS3BucketName(String bucketName) {
+  final checkBucket = Process.runSync(
+    'aws',
+    [
+      '--profile=${Platform.environment['AWS_PROFILE'] ?? 'default'}',
+      's3api',
+      'list-buckets',
+      '--query',
+      'Buckets[].Name',
+      '--output',
+      'text',
+    ],
+    stdoutEncoding: utf8,
+    stderrEncoding: utf8,
+  );
+
+  if (checkBucket.exitCode != 0) {
+    throw Exception(
+      '❌ Error checking if S3 bucket $bucketName exists: '
+      '${checkBucket.stdout}\n${checkBucket.stderr}',
+    );
+  }
+  String output = checkBucket.stdout as String;
+
+  /// Determine if bucket exists while ignoring the UUID
+  final bucketNameWithoutUUID = bucketName.substring(0, bucketName.length - 8);
+  final uuidMatcher = '[a-f0-9]{8}\$';
+  final pattern = '($bucketNameWithoutUUID)$uuidMatcher';
+
+  List<String> bucketNames = output.split('\t').map((e) => e.trim()).toList();
+  RegExp regex = RegExp(pattern);
+  String matchingBuckets = bucketNames.firstWhere(
+    (name) => regex.hasMatch(name),
+    orElse: () => '',
+  );
```

Comment:
Q: This will return an empty string if there is no bucket match. It will return the first bucket if there are multiple matches. Is that the desired behavior?

You could update to the following if you want to thrown an exception in either of those cases.

```
  final matchingBuckets = bucketNames.where((name) => regex.hasMatch(name));
  if (matchingBuckets.length != 1) {
    throw Exception(...);
  }
  return matchingBuckets.single;
```

Reply:
Good suggestion! Cleaned this up using your approach. We only want to throw if there are more than 1 bucket. If there isn't a bucket we want to create one.